misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -96,135 +97,149 @@
+                        lftTypeSymbol = findSymbol(leftTypeSymbols);
+                        foundTypeLeft = lftTypeSymbol.name();
+                        foundErrorLeft = true;
+                    }
+                    // If the check or checkpanic is used
+                    if (foundTypeRight.equals("")) {
+                        List<TypeSymbol> rightTypeSymbols = ((UnionTypeSymbol) rhsTypeSymbol).memberTypeDescriptors();
+                        rhsTypeSymbol = findSymbol(rightTypeSymbols);
+                        foundTypeRight = rhsTypeSymbol.name();
+                        foundErrorRight = true;
+                    }
+
+
+                    String functionCall = context.positionDetails().matchedNode().toString();
+                    if (foundErrorRight && !foundErrorLeft) {
+                        symbolAtCursorName = functionCall.split("[=;]")[1].trim();
+                        generatedFunctionName =
+                                String.format("map%sTo%s(check %s)", foundTypeRight, foundTypeLeft, symbolAtCursorName);
+                        fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
+                    } else if (foundErrorLeft && foundErrorRight) {
+                        // get the information about the line positions
+                        newTextRange = CommonUtil.toRange(context.positionDetails().matchedNode().lineRange());
+                        generatedFunctionName =
+                                String.format("map%sTo%s(%s)", foundTypeRight, foundTypeLeft, functionCall);
+                        fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
+                    } else {
+                        symbolAtCursorName = functionCall.split("[=;]")[1].trim();
+                        generatedFunctionName =
+                                String.format("map%sTo%s(%s)", foundTypeRight, foundTypeLeft, symbolAtCursorName);
+                        fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
+                    }
                     break;
-                } else {
-                    String functionParameters = functionCall.split("[()]")[1];
-                    generatedFunctionName =
-                            String.format("map%sTo%s(%s(%s))", foundTypeRight, foundTypeLeft, symbolAtCursorName,
-                                    functionParameters);
-                    fEdits.add(new TextEdit(newTextRange, generatedFunctionName));
+                default:
+                    throw new IllegalStateException("Unexpected value: " + symbolAtCursorType);
+            }
+
+            // Insert function declaration at the bottom of the file
+            String functionName = String.format("map%sTo%s", foundTypeRight, foundTypeLeft);
+            boolean found = fileContentSymbols.stream().anyMatch(p -> p.name().contains(functionName));
+            if (!found) {
```

Comment:
Can we reverse this logic to return `if(found)` logic first

Reply:
I made the change according to the comment. But IntelliJ Suggest me to extract the common parts.

Reply:
Issue here is if-block is too lengthy and we tend to lost the the context when reading the code. anyway that's a minor issue can improve in a separate PR